code/ship/ship.cpp

```diff
@@ -15619,9 +15619,32 @@ SCP_string ship_return_orders(ship* sp)
 // of what a ship's orders are.  Feel free to use this function if 
 // it suits your needs for something.
 char *ship_return_time_to_goal(char *outbuf, ship *sp)
+{
+	int	minutes, seconds;
+	int time = ship_return_seconds_to_goal(sp);
+	if ( time >= 0 ) {
+		minutes = time/60;
+		seconds = time%60;
+		if ( minutes > 99 ) {
+			minutes = 99;
+			seconds = 99;
+		}
+		sprintf(outbuf, NOX("%02d:%02d"), minutes, seconds);
+	} else if ( time == -2) {
+        // -2 is used as a signal of "there is no valid goal right now"
```

Comment:
This comment about -2 isn't necessary here, I don't think.  More importantly, this `else if` block should be removed because a message needs to go into `outbuf` even in this situation.